returnn/tf/layers/basic.py

```diff
@@ -859,39 +859,44 @@ class SliceNdLayer(_ConcatInputLayer):
   def __init__(self, start, size, min_size=None, **kwargs):
     """
     :param LayerBase start: (B,...)
-    :param int|None size: if None, it uses the max possible size, and it becomes a dynamic axis
+    :param int|LayerBase|None size: if None, it uses the max possible size,
+    and it becomes a dynamic axis. If LayerBase, it needs to have the same shape as start
```

Comment:
```suggestion
      and it becomes a dynamic axis
```
Indent by two spaces.

Remove the layer comment. This should not be the case. Use `get_common_data` on `start`/`size` and then `copy_compatible_to`.